saltgui/static/scripts/routes/page.js

```diff
@@ -335,6 +336,9 @@ class PageRoute extends Route {
     menu.addMenuItem("Show&nbsp;details", function(evt) {
       window.location.assign("/job?id=" + encodeURIComponent(job.id));
     }.bind(this));
+    menu.addMenuItem("Re-run&nbsp;job...", function(evt) {
+      this._runFullCommand(evt, job["Target-type"], job.Target, functionText);
```

Comment:
can you intercept the value "unknown-target".
In that case, Target-type should be default and Target should be empty.
Note that the command parameters are also lost in that case

Reply:
Ok I will try to handle "unknown-target" case and use default values it such situation.